morepath/tests/test_security.py

```diff
@@ -81,13 +79,119 @@ def remember(self, response, request, identity):
         def forget(self, response, request):
             pass
 
+    app.commit()
+
     c = Client(app())
 
     response = c.get('/foo')
     assert response.body == b'Model: foo'
     response = c.get('/bar', status=403)
 
 
+def test_permission_directive_identity_app_arg():
+    class App(morepath.App):
+        pass
+
+    class Model(object):
+        def __init__(self, id):
+            self.id = id
+
+    class Permission(object):
+        pass
+
+    @App.verify_identity()
+    def verify_identity(identity):
+        return True
+
+    @App.path(model=Model, path='{id}',
+              variables=lambda model: {'id': model.id})
+    def get_model(id):
+        return Model(id)
+
+    @App.permission_rule(model=Model, permission=Permission)
+    def get_permission(identity, model, permission):
+        if model.id == 'foo':
+            return True
+        else:
+            return False
+
+    @App.view(model=Model, permission=Permission)
+    def default(self, request):
+        return "Model: %s" % self.id
+
+    @App.identity_policy()
+    class IdentityPolicy(object):
+        def identify(self, app, request):
+            assert isinstance(app, App)
+            return Identity('testidentity')
+
+        def remember(self, app, response, request, identity):
```

Comment:
Hm, good point. I will remove this feature from these identity policy methods.

The notion of an identity policy as a collection of methods is one that is unique in Morepath, and it gives one to think: would it be better to ask developers to use 3 directives instead to install this stuff?